File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_03_viewmodel/domain/MomentRequest.java


package com.kunminx.jetpack_java.sample_03_viewmodel.domain;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.kunminx.jetpack_java.sample_03_viewmodel.data.DataRepository;
import com.kunminx.jetpack_java.common_data.bean.Moment;

import java.util.List;

/**
 * Create by KunMinX at 2020/5/30
 */
public class MomentRequest implements Request.IMomentRequest {

    private MutableLiveData<List<Moment>> mListMutableLiveData;

    @Override
    public LiveData<List<Moment>> getListMutableLiveData() {
        if (mListMutableLiveData == null) {
            mListMutableLiveData = new MutableLiveData<>();
        }
        return mListMutableLiveData;
    }

    @Override
    public void requestList() {
        DataRepository.getInstance().requestList(mListMutableLiveData);
    }
}

File: jetpack_kotlin/src/main/java/com/flywith24/jetpack_kotlin/sample_03_viewmodel/domain/MomentRequest.kt
package com.flywith24.jetpack_kotlin.sample_03_viewmodel.domain

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.flywith24.jetpack_kotlin.common_data.bean.Moment
import com.flywith24.jetpack_kotlin.sample_03_viewmodel.data.DataRepository

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   22:09
 * description
 */
internal class MomentRequest : Request.IMomentRequest {
    private val mListMutableLiveData = MutableLiveData<List<Moment>>()

    override fun getListMutableLiveData(): LiveData<List<Moment>> = mListMutableLiveData

    override fun requestList() {
        DataRepository.getInstance().requestList(mListMutableLiveData)
    }
}